src/App.js

```diff
@@ -76,6 +80,40 @@ export default class App extends Component {
     this.setState({characterName: event.target.value})
   }
 
+  updateAbilityScore (ability, score) {
+    let newScores = this.state.activeCharacter.hasOwnProperty('abilityScores')
+      ? Object.assign(this.state.activeCharacter.abilityScores, {[ability]: score})
+      : Object.assign({STR: 8, DEX: 8, CON: 8, INT: 8, WIS: 8, CHA: 8}, {[ability]: score})
+
+    const newActiveCharacter = Object.assign(this.state.activeCharacter, {abilityScores: newScores})
+    this.setState({activeCharacter: newActiveCharacter})
+
+    const putData = {
+      method: 'PUT',
+      body: JSON.stringify(this.state.activeCharacter)
+    }
+
+    fetch(`${SERVER_ROOT}/characters/${this.state.activeCharacterId}.json`, putData)
+      .catch((error) => {
+        console.log(error)
+      })
+  }
+
+  updateActiveCharacter (uid) {
+    this.setState({activeCharacterId: uid})
+
+    fetch(`${SERVER_ROOT}/characters/${uid}.json`)
+      .then((response) => {
+        return response.json()
+      })
+      .then((json) => {
+        this.setState({activeCharacter: json})
+      })
+      .catch((error) => {
+        console.log(error)
```

Comment:
Might be helpful at some point soon to have a "Display errors in a friendly way to the user" story, because unless the user has their console open, they may never know that none of their work is getting saved!